modules/repository-s3/build.gradle

```diff
@@ -182,197 +179,35 @@ tasks.named("internalClusterTest").configure {
-    // a regional endpoint without an error. It would be great to make actual requests against
-    // a test fixture, but setting the region means using a production endpoint
-    systemProperty 'tests.rest.blacklist', [
-      'repository_s3/20_repository_permanent_credentials/*',
-      'repository_s3/30_repository_temporary_credentials/*',
-      'repository_s3/40_repository_ec2_credentials/*',
-      'repository_s3/50_repository_ecs_credentials/*',
-      'repository_s3/60_repository_sts_credentials/*'
-    ].join(",")
-  }
-  tasks.named("check").configure { dependsOn("yamlRestTestRegionalSTS") }
-
-  testClusters.matching { it.name == "yamlRestTestRegionalSTS" }.configureEach {
-    module tasks.named("explodedBundlePlugin")
-
-    File awsWebIdentityTokenExternalLocation = file('src/test/resources/aws-web-identity-token-file')
-    extraConfigFile 'repository-s3/aws-web-identity-token-file', awsWebIdentityTokenExternalLocation
-    environment 'AWS_WEB_IDENTITY_TOKEN_FILE', "$awsWebIdentityTokenExternalLocation"
-    environment 'AWS_ROLE_ARN', 'arn:aws:iam::123456789012:role/FederatedWebIdentityRole'
-    environment 'AWS_ROLE_SESSION_NAME', 'sts-fixture-test'
-    // Force the repository to set a regional production endpoint
-    environment 'AWS_STS_REGIONAL_ENDPOINTS', 'regional'
-    environment 'AWS_REGION', 'ap-southeast-2'
-  }
+  // ideally we could resolve an env path in cluster config as resource similar to configuring a config file
+  // not sure how common this is, but it would be nice to support
+  File awsWebIdentityTokenExternalLocation = file('src/test/resources/aws-web-identity-token-file')
+  // The web identity token can be read only from the plugin config directory because of security restrictions
+  // Ideally we would create a symlink, but extraConfigFile doesn't support it
+  nonInputProperties.systemProperty("awsWebIdentityTokenExternalLocation", awsWebIdentityTokenExternalLocation.getAbsolutePath())
 }
 
 // 3rd Party Tests
-TaskProvider s3ThirdPartyTest = tasks.register("s3ThirdPartyTest", Test) {
+tasks.register("s3ThirdPartyTest", Test) {
   SourceSetContainer sourceSets = project.getExtensions().getByType(SourceSetContainer.class);
   SourceSet internalTestSourceSet = sourceSets.getByName(InternalClusterTestPlugin.SOURCE_SET_NAME)
   setTestClassesDirs(internalTestSourceSet.getOutput().getClassesDirs())
   setClasspath(internalTestSourceSet.getRuntimeClasspath())
   include '**/S3RepositoryThirdPartyTests.class'
+  // test container accesses ~/.testcontainers.properties read
+  systemProperty "tests.security.manager", "false"
   systemProperty 'test.s3.account', s3PermanentAccessKey
   systemProperty 'test.s3.key', s3PermanentSecretKey
   systemProperty 'test.s3.bucket', s3PermanentBucket
   nonInputProperties.systemProperty 'test.s3.base', s3PermanentBasePath + "_third_party_tests_" + BuildParams.testSeed
-  if (useFixture) {
```

Comment:
So I think one side-effect of this is we no longer have the ability to run the third party tests against a fixture. We can _only_ run these with a real external service. I think we should wire the test case up to still support using fixtures. This is the case in the other repository modules and is helpful for local development.

Reply:
I think this was never wired against the s3 fixtures. it just uses the minio fixture. Originally its configuration was not within that if statement: https://github.com/elastic/elasticsearch/blob/main/modules/repository-s3/build.gradle#L360C1-L375

Reply:
Yes, because it probably only needed the minio fixture to run. Minio is just a mock S3 implementation. You can see below I can run these tests locally. I'm pretty sure that would fail with this PR.

https://gradle-enterprise.elastic.co/s/dfnnlb42ib4wc/tests/overview